### PR TITLE
feat(message): send message flag

### LINKED ISSUE #978: Add support for setting flags when sending a message
### Summary

The library currently doesn't support setting Message flags

### What is the feature request for?

The core library

### The Problem

Link to API Docs: https://discord.com/developers/docs/resources/channel#create-message-jsonform-params

As you can see in the above linked docs, a field called "flags" is shown but the library currently doesn't support that field.

### The Ideal Solution

Preferably a `flags` kwarg that takes a list of `MessageFlags` should be added to `Messageable.send` and locked down to the currently supported flags like `SUPPRESS_EMBEDS` or an user friendlier solution would be a `suppress_embeds` kwarg that takes a bool and defaults to False.

### The Current Solution

N/A

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MaskDuck**
## Summary

closes #978 

## Checklist

- [ ] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by ooliver1 — COMMENTED** [review]
Test your own PRs and fix the pyright issues once again please.

**Review by teaishealthy — DISMISSED** [review]
untested

File: nextcord/abc.py [hunk 1]
```diff
@@ -1318,6 +1326,8 @@ async def send(
         reference: Optional[Union[Message, MessageReference, PartialMessage]] = None,
         mention_author: Optional[bool] = None,
         view: Optional[View] = None,
+        flags: Optional[MessageFlags] = None,
+        suppress_embeds: Optional[bool] = None,
```
Comment (EmreTech) [inline nextcord/abc.py:1330]: Why `Optional[bool]`? Why not just make the default `False`?

File: nextcord/abc.py [hunk 2]
```diff
@@ -1411,6 +1432,12 @@ async def send(
         channel = await self._get_channel()
         state = self._state
         content = str(content) if content is not None else None
+        if flags is None:
+            flags = MessageFlags()
+        if suppress_embeds is True:
+            flags.suppress_embeds = True
```
Comment (EmreTech) [inline nextcord/abc.py:1438]: 1. `if x is True` violates PEP 8 recommendations.
2. Accompanied with my previous review comment, why not just set `flags.suppress_embeds` directly to `suppress_embeds`?

File: nextcord/abc.py (hunk 1 above)
Comment (ooliver1) [reply nextcord/abc.py:1330]: Same reason as `mention_author`, default to the API default.

File: nextcord/abc.py (hunk 2 above)
Comment (ooliver1) [reply nextcord/abc.py:1438]: This should instead simply check for `None` then pass it explicitly, as this relies on the API default - consistency.
```suggestion
        if suppress_embeds is not None:
            flags.suppress_embeds = suppress_embeds
```

File: nextcord/abc.py [hunk 3]
```diff
@@ -1388,6 +1398,16 @@ async def send(
...
 
             .. versionadded:: 2.0
+        flags: Optional[:class:`~nextcord.MessageFlags`]
+            The message flags being set for this message.
+            Currently only :class:`~nextcord.MessageFlags.suppress_embeds` is able to be set.
+
+            .. versionadded:: 2.4
+        suppress_embeds: Optional[:class:`bool`]
+            Whether to suppress embeds on this message.
+            This cannot be set with ``flags``.
```
Comment (Skelmis) [inline nextcord/abc.py:1408]: Looks like it can be set in conjunction though?

Comment (ooliver1) [reply nextcord/abc.py:1408]: I don't even see much point in `flags`, since if a flag isnt implemented, you can't really send it.

Comment (Soheab) [reply nextcord/abc.py]: it allows for future proofing I guess

**Review by ooliver1 — COMMENTED** [review]
This seems to be settable when creating a forum post.

**Review by ooliver1 — APPROVED** [review]

File: nextcord/abc.py [hunk 4]
```diff
@@ -1400,7 +1419,8 @@ async def send(
             you specified both ``file`` and ``files``,
             or you specified both ``embed`` and ``embeds``,
             or the ``reference`` object is not a :class:`~nextcord.Message`,
-            :class:`~nextcord.MessageReference` or :class:`~nextcord.PartialMessage`.
+            :class:`~nextcord.MessageReference` or :class:`~nextcord.PartialMessage`,
+            or you specified both ``flags`` and ``suppress_embeds``.
```
Comment (EmreTech) [inline nextcord/abc.py:1423]: `nextcord.InvalidArgument` isn't raised for this case?

File: nextcord/webhook/async_.py [hunk 5]
```diff
@@ -481,6 +481,9 @@ def handle_message_parameters(
     view: Optional[View] = MISSING,
     allowed_mentions: Optional[AllowedMentions] = MISSING,
     previous_allowed_mentions: Optional[AllowedMentions] = None,
+    ephemeral: Optional[bool] = None,
+    flags: Optional[MessageFlags] = None,
```
Comment (EmreTech) [inline nextcord/webhook/async_.py:485]: Shouldn't an exception be raised for when `suppress_embeds`/`ephemeral` are mixed with `flags`?

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: nextcord/webhook/async_.py (hunk 5 above)
Comment (EmreTech) [reply nextcord/webhook/async_.py:485]: What do you mean?

Comment (ooliver1) [reply nextcord/webhook/async_.py:485]: ephemeral just overrides flags

**Review by EmreTech — APPROVED** [review]

**Pull request merged**